Applications/Spire/Source/KeyBindings/TimeInForceItemDelegate.cpp

```diff
@@ -0,0 +1,73 @@
+#include "Spire/KeyBindings/TimeInForceItemDelegate.hpp"
+#include "Nexus/Definitions/TimeInForce.hpp"
+#include "Spire/KeyBindings/InputFieldEditor.hpp"
+
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+namespace {
+  auto to_qstring(TimeInForce::Type type) {
+    return QString::fromStdString(ToString(type));
+  }
+
+  auto create_time_in_force_item_list() {
+    static auto list = std::vector<QString>({
+        to_qstring(TimeInForce::Type::DAY),
+        to_qstring(TimeInForce::Type::FOK),
+        to_qstring(TimeInForce::Type::GTC),
+        to_qstring(TimeInForce::Type::GTD),
+        to_qstring(TimeInForce::Type::GTX),
+        to_qstring(TimeInForce::Type::IOC),
+        to_qstring(TimeInForce::Type::MOC),
+        to_qstring(TimeInForce::Type::OPG)
+      });
+    return list;
+  }
+
+  auto get_time_in_force_variant(const QString& text) {
+    if(text == to_qstring(TimeInForce::Type::DAY)) {
```

Comment:
for(auto type : { TimeInForce::Type::DAY, TimeInForce::Type::FOK, ... }) {
      if(text == to_qstring(type)) {
        return QVariant::fromValue<TimeInForce>({type});
      }
    }